massgen/orchestrator.py

```
@@ -13429,6 +13442,546 @@ async def _spawn_trace_analyzer_background(
+        self._criteria_evolution_completed_labels.add(label_tuple)
+
+        # Re-initialize checklist tool with new criteria
+        try:
+            self._init_checklist_tool()
+        except Exception:
+            logger.warning("[Orchestrator] Failed to re-init checklist tool after criteria evolution", exc_info=True)
+
+        # Emit event + push updated criteria to TUI
+        try:
+            _emitter = get_event_emitter()
+            evolved_count = sum(1 for old_c, new_c in zip(old_criteria, evolved) if old_c.text != new_c.text)
```

Ruff (0.15.9)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)